src/main/kotlin/mate/academy/service/FigureSupplier.kt

```diff
@@ -0,0 +1,62 @@
+package mate.academy.service
+
+import mate.academy.model.Figure
+import mate.academy.model.FigureType
+import mate.academy.model.Color
+import mate.academy.model.Circle
+import mate.academy.model.Rectangle
+import mate.academy.model.RightTriangle
+import mate.academy.model.IsoscelesTrapezoid
+import mate.academy.model.Square
+import kotlin.random.Random
+
+private const val MIN_SIDE: Int = 10
+private const val MAX_SIDE: Int = 100
+private const val DEFAULT_RADIUS: Int = 10
+
+class FigureSupplier {
+    private val colorSupplier = ColorSupplier()
+
+    fun getRandomFigure(): Figure {
+        val randomColor = colorSupplier.getRandomColor()
+        val randomFigureType = FigureType.values().random()
+        return when (randomFigureType) {
+            FigureType.SQUARE -> generateRandomSquare(randomColor)
+            FigureType.CIRCLE -> generateRandomCircle(randomColor)
+            FigureType.RECTANGLE -> generateRandomRectangle(randomColor)
+            FigureType.RIGHT_TRIANGLE -> generateRandomRightTriangle(randomColor)
+            FigureType.ISOSCELES_TRAPEZOID -> generateRandomIsoscelesTrapezoid(randomColor)
+        }
+    }
+
+    private fun generateRandomSquare(color: Color): Figure = Square(

```

Comment:
This code will work, however there is some design limitation.
As this method and similar ones could return only random Figure.
More flexible way is to use methods which accept Figures sides, ect... as parameters.
In this case you may reuse this code for creating both random and exact Figures